kratos/utilities/entities_utilities.h

```diff
@@ -32,6 +33,148 @@ namespace Kratos
+    {
+    public:
+        ///@name Type Definitions
+        ///@{
+
+        /// Geometry type definition
+        using GeometryType = typename TEntity::GeometryType;
+
+        /// Pointer definition of ReplaceElementsAndConditionsProcess
+        KRATOS_CLASS_POINTER_DEFINITION(EntitityIdentifier);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+            
+        /** 
+         * @brief Default constructor 
+         */
+        EntitityIdentifier() :
+            mName("")
+        {
+        }
+
+        /**
+         * @brief Constructor
+         * @param rName The name of the entity.
+         */
+        EntitityIdentifier(const std::string& rName);
+
+        /**
+         * @brief Copy constructor
+         */
+        EntitityIdentifier(const EntitityIdentifier& rOther)
+            : mpPrototypeEntity(rOther.mpPrototypeEntity),
+              mDefinitionType(rOther.mDefinitionType),
+              mName(rOther.mName),
+              mTypes(rOther.mTypes)
+        {
+        }
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+        /**
+         * @brief Assignment operator.
+         */
+        EntitityIdentifier& operator=(const EntitityIdentifier& rOther)
+        {
+            mpPrototypeEntity = rOther.mpPrototypeEntity;
+            mDefinitionType = rOther.mDefinitionType;
+            mName = rOther.mName;
+            mTypes = rOther.mTypes;
+
+            return *this;
+        }
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        /**
+         * @brief Checks if the object is initialized.
+         * @return true if the object is initialized, false otherwise.
+         */
+        bool IsInitialized();
+
+        /**
+         * @brief Get the prototype entity.
+         * @param pGeometry The pointer to the geometry.
+         * @return const TEntity& The prototype entity.
+         */
+        const TEntity& GetPrototypeEntity(typename GeometryType::Pointer pGeometry);
+
+        ///@}
+        ///@name Input and output
+        ///@{
+
+        /// Turn back information as a string.
+        std::string Info() const
+        {
+            return "EntitityIdentifier";
+        }
+
+        /// Print information about this object.
+        void PrintInfo(std::ostream& rOStream) const
+        {
+            rOStream << "EntitityIdentifier";
+        }
+
+        /// Print object's data.
+        void PrintData(std::ostream& rOStream) const;
+
+        ///@}
+    private:
+        ///@name Member Variables
+        ///@{
+
+        typename TEntity::Pointer mpPrototypeEntity = nullptr;                    /// The prototype entity
+        DefinitionType mDefinitionType = DefinitionType::Single;                  /// The type of definition
+        std::string mName;                                                        /// The name of the entity
+        std::unordered_map<GeometryData::KratosGeometryType, std::string> mTypes; /// The settings of the entities
```

Comment:
do you need to store the name of the entity? wouldn't it be better to store a pointer to the reference element/condition with that type?

in any case, let's imagine we want to apply two elements (or one element and a condition to the same geometry type). What would we do in that case?

Reply:
If you see the implementation you will see it is required for the templated case